src/utils/popup.js

```diff
@@ -0,0 +1,45 @@
+import {escape} from "../const/const";
+
+export default class Popup {
+
+    _isOpenPopup = false;
+
+    constructor(popupSelector) {
+        this.popupElement = document.querySelector(popupSelector);
+    }
+
+    _handleEscClose = (evt) => {
+        if (evt.key === escape) {
+            this.close();
+        }
+    };
+
+    _handelMouseDownClose = (evt) => {
+        if (!evt.target.classList.contains('popup_opened') && !evt.target.classList.contains('attachment')
+            && !evt.target.offsetParent.classList.contains('popup_opened')) {
```

Comment:
это условие выглядит очень сложно и при первом взгляде на него, придется приложить ментальные усилия, чтобы его понять. Хороший код == простой код. Как ты смотришь на такой рефакторинг?
```suggestion
        const popupNotOpened = !evt.target.classList.contains('popup_opened')
        const hasNoAttachments = !evt.target.classList.contains('attachment')
        const parentPopupNotOpened = !evt.target.offsetParent.classList.contains('popup_opened')
        if (popupNotOpened && hasNoAttachments && parentPopupNotOpened) {
```

Reply:
Да, согласен!